shared/indexer/indexer_test.go

```diff
@@ -0,0 +1,269 @@
+	require.True(t, txResultsEqual(t, txResult, txResultFromHash))
+	// check indexing/get by hash 2
+	hash2, err := txResult2.Hash()
+	require.NoError(t, err)
+	txResultFromHash2, err := txIndexer.GetByHash(hash2)
+	require.NoError(t, err)
+	require.True(t, txResultsEqual(t, txResult2, txResultFromHash2))
+}
+
+func TestGetByHeight(t *testing.T) {
+	txIndexer, err := NewMemTxIndexer()
+	defer txIndexer.Close()
+	// setup 3 transactions
+	txResult := NewTestingTransactionResult(t, 0, 0)
+	require.NoError(t, err)
+	txResult2 := NewTestingTransactionResult(t, 0, 1)
+	require.NoError(t, err)
+	txResult3 := NewTestingTransactionResult(t, 1, 0)
+	require.NoError(t, err)
+	// index all 3 transactions
+	err = txIndexer.Index(txResult)
+	require.NoError(t, err)
+	err = txIndexer.Index(txResult2)
+	require.NoError(t, err)
+	err = txIndexer.Index(txResult3)
+	require.NoError(t, err)
+	// check indexing/get by height
+	txResultsFromHeight, err := txIndexer.GetByHeight(0, false)
+	require.NoError(t, err)
+	txResultsFromHeight1, err := txIndexer.GetByHeight(1, false)
+	require.NoError(t, err)
+	require.Equal(t, 2, len(txResultsFromHeight))
+	require.Equal(t, 1, len(txResultsFromHeight1))
+}
+
+func TestGetBySender(t *testing.T) {
+	txIndexer, err := NewMemTxIndexer()
+	defer txIndexer.Close()
+	// setup transaction
+	txResult := NewTestingTransactionResult(t, 1, 0)
+	require.NoError(t, err)
+	// index transaction
+	err = txIndexer.Index(txResult)
+	require.NoError(t, err)
+	// check indexing by sender / recipient
+	sender := txResult.GetSignerAddr()
+	require.NoError(t, err)
+	txResultsFromSender, err := txIndexer.GetBySender(sender, false)
+	require.NoError(t, err)
+	require.NoError(t, err)
+	require.Equal(t, 1, len(txResultsFromSender))
+	require.True(t, txResultsEqual(t, txResult, txResultsFromSender[0]))
+	// ensure it's not indexed elsewhere
+	txResultsFromRecipientBad, err := txIndexer.GetByRecipient(sender, false)
+	require.NoError(t, err)
+	require.Equal(t, 0, len(txResultsFromRecipientBad))
+}
+
+func TestGetByRecipient(t *testing.T) {
+	txIndexer, err := NewMemTxIndexer()
+	defer txIndexer.Close()
+	// setup tx
+	txResult := NewTestingTransactionResult(t, 1, 0)
+	require.NoError(t, err)
+	// index transactions
+	err = txIndexer.Index(txResult)
+	require.NoError(t, err)
+	recipient := txResult.GetRecipientAddr()
+	require.NoError(t, err)
+	txResultsFromRecipient, err := txIndexer.GetByRecipient(recipient, false)
+	require.NoError(t, err)
+	require.Equal(t, 1, len(txResultsFromRecipient))
+	require.True(t, txResultsEqual(t, txResult, txResultsFromRecipient[0]))
+	// ensure it's not indexed elsewhere
+	txResultsFromSenderBad, err := txIndexer.GetBySender(recipient, false)
+	require.NoError(t, err)
+	require.Equal(t, 0, len(txResultsFromSenderBad))
+}
+
+func txResultsEqual(t *testing.T, txR1, txR2 TxResult) bool {
+	bz, err := txR1.Bytes()
+	require.NoError(t, err)
+	bz2, err := txR2.Bytes()
+	require.NoError(t, err)
+	return bytes.Equal(bz, bz2)
```

Comment:
You can write `require.Equal(bz, bz2)`. If it's not equal, it'll output the diff too.

Will help remove `require.True` elsewhere as well.